Improve caching docs (#597)
RevenueCat/revenuecat-docs: docs_source/Test - Launch/debugging/caching.md

The latest `CustomerInfo` is automatically fetched and cached when the _Purchases SDK_ is configured and throughout the lifecycle of your app, so in most cases the `getCustomerInfo()` method will return synchronously. It is safe to call `getCustomerInfo()`  as often as needed and is a good idea to call it any time a user accesses premium content.

The SDK will update the cache after:

- 5 minutes with the app in the foreground
- 25 hours with the app in the background
- a purchase is made or restored

> For faster `customerInfo` updates, we recommed enabling [Platform Server Notifications](https://www.revenuecat.com/docs/server-notifications) and the `CustomerInfo` [listener](https://www.revenuecat.com/docs/configuring-sdk#advanced-configuration). 

Reviewer: Nit:
```suggestion
For faster `CustomerInfo` updates, we recommed enabling [Platform Server Notifications](https://www.revenuecat.com/docs/server-notifications) and the `CustomerInfo` [listener](https://www.revenuecat.com/docs/configuring-sdk#advanced-configuration). 
```

> For faster `CustomerInfo` updates, we recommed enabling [Platform Server Notifications](https://www.revenuecat.com/docs/server-notifications) and the `CustomerInfo` [listener](https://www.revenuecat.com/docs/configuring-sdk#advanced-configuration). 

Reviewer 2: ```suggestion
For faster `CustomerInfo` updates, we recommend enabling [Platform Server Notifications](https://www.revenuecat.com/docs/server-notifications) and the `CustomerInfo` [listener](https://www.revenuecat.com/docs/configuring-sdk#advanced-configuration). 
```

> See here for information on the [CustomerInfo's fetchPolicy](https://revenuecat.github.io/purchases-ios-docs/4.19.0/documentation/revenuecat/purchases/customerinfo(fetchpolicy:)/).

Reviewer: It's unfortunate that we don't have a way to point to the latest version of a particular file. Let's maybe use a more recent version at least:
```suggestion
See here for information on the [CustomerInfo's fetchPolicy](https://revenuecat.github.io/purchases-ios-docs/4.31.5/documentation/revenuecat/purchases/customerinfo(fetchpolicy:)/).
```
